src/main/java/net/datafaker/transformations/sql/SqlTransformer.java

```diff
@@ -87,69 +87,59 @@ public CharSequence apply(IN input, Schema<IN, ?> schema, int rowId) {
         if (fields.length == 0) {
             return EMPTY_RESULT;
         }
-        StringBuilder sb = new StringBuilder();
         if (withBatchMode) {
             if (rowId == 0 || batchSize > 0 && rowId % batchSize == 0) {
-                sb.append(
-                    SqlDialect.getFirstRow(
-                        dialect, () -> appendTableInfo(fields), () -> addValues(input, fields), keywordCase));
+                return SqlDialect.getFirstRow(
+                        dialect, () -> appendTableInfo(fields), () -> addValues(input, fields), keywordCase);
             } else {
-                sb.append(",")
-                    .append(LINE_SEPARATOR)
-                    .append(
-                        SqlDialect.getOtherRow(
-                            dialect, () -> appendTableInfo(fields), () -> addValues(input, fields), keywordCase));
+                return "," + LINE_SEPARATOR
+                        + SqlDialect.getOtherRow(
+                        dialect, () -> appendTableInfo(fields), () -> addValues(input, fields), keywordCase);
             }
         } else {
-            sb.append(INSERT_INTO.getValue(keywordCase))
-                .append(" ")
-                .append(appendTableInfo(fields))
-                .append(" ")
-                .append(VALUES.getValue(keywordCase))
-                .append(" ")
-                .append(addValues(input, fields));
+            return INSERT_INTO.getValue(keywordCase)
+                    + " "
+                    + appendTableInfo(fields)
+                    + " "
+                    + VALUES.getValue(keywordCase)
+                    + " "
+                    + addValues(input, fields);
         }
 
-        return sb.toString();
     }
```

Comment:
I think in this particular part using the String builder might actually be better vs concatenation.

Maybe the original instantiation should specify a larger initial capacity (IIRC default is 16 char).

Reply:
yes, you're right
there is another way: use of `String.join` which does not have the issue of rebuilding an array...
however there is an issue of vararg... 
anyway sith `String.join` it is still faster than the initial version